Libraries/Geolocation/RCTLocationObserver.m

```diff
@@ -128,7 +134,7 @@ - (void)beginLocationUpdates
 {
   if (!_locationManager) {
     _locationManager = [CLLocationManager new];
-    _locationManager.distanceFilter = RCT_DEFAULT_LOCATION_ACCURACY;
+    _locationManager.distanceFilter = _observerOptions.distanceFilter;
```

Comment:
I just tested calling `getCurrentPosition` from the start and it seems that it does not crash because structs have a default value:
<img width="961" alt="screen shot 2016-04-14 at 8 44 09 pm" src="https://cloud.githubusercontent.com/assets/159813/14550911/06322072-0282-11e6-8e47-3fdf3e067f6f.png">
I think using `RCT_DEFAULT_LOCATION_ACCURACY` or `kCLDistanceFilterNone` should be used here.